bin/ons_csv_to_ctb_json_ds_vars.py

```diff
@@ -16,71 +17,81 @@ def __init__(self, dataset_mnemonic, filename, all_classifications):
         """Add non-geographic variable ensuring data integrity."""
         variable_mnemonic = variable['Variable_Mnemonic']
         classification_mnemonic = variable['Classification_Mnemonic']
         if not classification_mnemonic:
-            raise ValueError(f'Reading {self.filename} '
-                             'Classification must be specified for non-geographic '
-                             f'{variable_mnemonic} in dataset {self.dataset_mnemonic}')
+            self.recoverable_error(f'Reading {self.filename}:{row_num} '
+                                   'Classification must be specified for non-geographic '
+                                   f'{variable_mnemonic} in dataset {self.dataset_mnemonic}')
+            logging.warning(f'Reading {self.filename}:{row_num} dropping record')
+            return
+
         if variable['Lowest_Geog_Variable_Flag'] == 'Y':
-            raise ValueError(f'Reading {self.filename} '
-                             'Lowest_Geog_Variable_Flag set on non-geographic variable'
-                             f' {variable_mnemonic} for dataset {self.dataset_mnemonic}')
+            self.recoverable_error(f'Reading {self.filename}:{row_num} '
+                                   'Lowest_Geog_Variable_Flag set on non-geographic variable '
+                                   f'{variable_mnemonic} for dataset {self.dataset_mnemonic}')
 
         classification = self.all_classifications[classification_mnemonic]
         if classification.private['Variable_Mnemonic'] != variable_mnemonic:
-            raise ValueError(f'Reading {self.filename} Invalid '
-                             f'classification {classification_mnemonic} '
-                             f'specified for variable {variable_mnemonic} '
-                             f'in dataset {self.dataset_mnemonic}')
+            self.recoverable_error(f'Reading {self.filename}:{row_num} Invalid '
+                                   f'classification {classification_mnemonic} '
+                                   f'specified for variable {variable_mnemonic} '
+                                   f'in dataset {self.dataset_mnemonic}')
+            logging.warning(f'Reading {self.filename}:{row_num} dropping record')
+            return
+
         if not variable['Processing_Priority']:
-            raise ValueError(f'Reading {self.filename} '
-                             'Processing_Priority not specified for classification '
-                             f'{classification_mnemonic} in dataset '
-                             f'{self.dataset_mnemonic}')
+            self.recoverable_error(f'Reading {self.filename}:{row_num} '
+                                   'Processing_Priority not specified for classification '
+                                   f'{classification_mnemonic} in dataset '
+                                   f'{self.dataset_mnemonic}')
+            variable['Processing_Priority'] = 0
```

Comment:
Is it worth mentioning what the assumed value is going to be (e.g. 0 in this case)

Reply:
Fair point. Probably also worth saying that we're ignoring fields when there's an invalid optional field.